kafka-rest/src/main/java/io/confluent/kafkarest/response/StreamingResponseIdleTimeLimiter.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.kafkarest.response;
+
+import static java.util.Objects.requireNonNull;
+
+import com.google.common.util.concurrent.TimeLimiter;
+import io.confluent.kafkarest.exceptions.v3.ErrorResponse;
+import java.time.Duration;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+
+public final class StreamingResponseIdleTimeLimiter {
+  private final TimeLimiter delegate;
+  private final Duration maxIdleTime;
+
+  public StreamingResponseIdleTimeLimiter(TimeLimiter delegate, Duration maxIdleTime) {
+    this.delegate = requireNonNull(delegate);
+    this.maxIdleTime = requireNonNull(maxIdleTime);
+  }
+
+  <T> T callWithTimeout(Callable<T> callable)
+      throws ExecutionException, InterruptedException, MaxIdleTimeExceededException {
+    try {
+      return delegate.callWithTimeout(callable, maxIdleTime);
```

Comment:
If we are planning on having reslience4j as our rate limiter (due to it being able to deal with spikes of data better than guava), should we also use it here?